flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -794,23 +796,24 @@ public void testOperatorClosingBeforeStopRunning() throws Throwable {
 		}
 	}
 
-	/**
-	 * Test set user code ClassLoader before calling ProcessingTimeCallback.
-	 */
 	@Test
-	public void testSetsUserCodeClassLoaderForTimerThreadFactory() throws Throwable {
-		syncLatch = new OneShotLatch();
-
+	public void testThreadInvariants() throws Throwable {
```

Comment:
Again, why exactly if this changed if there is even a test to guard the old behavior? Did you talk to the author of the test if this is acceptable?

Reply:
I haven't talked with the author of the old test. But:
 1. the old test doesn't work anymore, as it deadlocks. This is expected, due to the change in this PR.
 2. the thing that was checked by the test, doesn't required anymore, as the timer thread is not execution user code anymore.

Above two reasons why the test could have been removed. But instead, I have modified it to tailor to the new situation with mailbox (I don't know if such check already exists).